### PR TITLE
ciao-controller: validate API docs with Travis

### PR DESCRIPTION
**Pull request by leoswaldo**
Automation of API Documentation.

CIAO-Controller has a API that exposes CIAO methods through HTTPS, documentation will be changing frequently due to the changes and new features in CIAO. 

Swagger will  allow us to automate the process of Docs generations. When we nave a new release the documentation will be updated, and every new PR needs to be validated it does not break the documentation.

This PR is to validate no issues with new integrations and the generation of docs for the API.

### CODE REVIEW

**Comment by sameo** [comment]
This looks good to me.

File: ciao-controller/compute.go [hunk 1]
```diff
@@ -1585,4 +1790,5 @@ func createComputeAPI(context *controller) {
 
 	service := fmt.Sprintf(":%d", computeAPIPort)
 	log.Fatal(http.ListenAndServeTLS(service, httpsCAcert, httpsKey, r))
+
```
Comment (sameo) [inline ciao-controller/compute.go]: Unneeded new line.

File: ciao-controller/compute.go [hunk 2]
```diff
@@ -1544,7 +1749,7 @@ func createComputeAPI(context *controller) {
 	}).Methods("GET")
 
 	r.HandleFunc("/v2.1/flavors/{flavor}/servers/detail", func(w http.ResponseWriter, r *http.Request) {
-		listServerDetails(w, r, context)
+		listFlavorServerDetail(w, r, context)
```
Comment (sameo) [inline ciao-controller/compute.go]: Why are we changing this ?

File: ciao-controller/compute.go [hunk 3]
```diff
@@ -1535,7 +1740,7 @@ func createComputeAPI(context *controller) {
 	}).Methods("GET")
 
 	r.HandleFunc("/v2.1/{tenant}/events", func(w http.ResponseWriter, r *http.Request) {
-		listEvents(w, r, context)
+		listTenantEvents(w, r, context)
```
Comment (sameo) [inline ciao-controller/compute.go]: Why are we changing this ?

File: ciao-controller/compute.go (hunk 2 above)
Comment (leoswaldo) [inline ciao-controller/compute.go]: Swagger documents API calls 1 per function, and there were two API calling the same method, that means that one call would not be documented, for this reason, I had to create "secondary functions" that call the primary. This way we can document all API calls

File: ciao-controller/compute.go (hunk 3 above)
Comment (leoswaldo) [inline ciao-controller/compute.go]: same case as previous one

File: ciao-controller/compute.go [hunk 4]
```diff
@@ -1434,6 +1609,28 @@ func listEvents(w http.ResponseWriter, r *http.Request, context *controller) {
...
+// @Failure 500 {object} payloads.HTTPReturnErrorCode "The response contains the corresponding message and 50x corresponding code."
+// @Router /v2.1/{tenant}/events [get]
+// @Resource /v2.1/events
+// listTenantEvents is created with the only purpose of API documentation for method
+// /v2.1/{tenant}/events
+func listTenantEvents(w http.ResponseWriter, r *http.Request, context *controller) {
+	listEvents(w, r, context)
+}
+
+// @Title listTenantEvents
```
Comment (sameo) [inline ciao-controller/compute.go]: Title is wrong it seems.

File: ciao-controller/compute.go (hunk 2 above)
Comment (sameo) [inline ciao-controller/compute.go]: Fair enough.

**Pull request merged**